firebase-app-distribution/src/main/java/com/google/firebase/appdistribution/FirebaseAppDistribution.java

```diff
@@ -141,16 +184,16 @@ public static FirebaseAppDistribution getInstance(@NonNull FirebaseApp app) {
    * sign in UI
    */
   @NonNull
-  public synchronized Task<AppDistributionRelease> checkForUpdate() {
+  public synchronized Task<AppDistributionRelease> checkForNewRelease() {
     if (cachedCheckForUpdateTask != null && !cachedCheckForUpdateTask.isComplete()) {
       LogWrapper.getInstance().v("Response in progress");
       return cachedCheckForUpdateTask;
     }
-
     cachedCheckForUpdateTask =
         signInTester()
-            .onSuccessTask(unused -> this.checkForUpdateClient.checkForUpdate())
+            .onSuccessTask(newReleaseExecutor, unused -> this.checkForUpdateClient.checkForUpdate())
```

Comment:
we may want to try removing these to see if the UI thread can handle this as well